pkg/agent/controller/egress/egress_controller.go

```diff
@@ -80,6 +86,8 @@ var emptyWatch = watch.NewEmptyWatch()
 
 var newIPAssigner = ipassigner.NewIPAssigner
 
+var egressIPMap map[string]bool
```

Comment:
We should never use global variable to store such information, which is bad for encapsulation and testing.

Reply:
okay will keep this in mind, and will make the required changes.